keanu-project/src/main/java/io/improbable/keanu/tensor/bool/BooleanTensor.java

```diff
@@ -31,6 +34,35 @@ static BooleanTensor falses(long... shape) {
         return new SimpleBooleanTensor(false, shape);
     }
 
+     /**
+     * @param dimension  the dimension along which toStack are stacked
+     * @param toStack    an array of BooleanTensor's of the same shape
+     * @return  a BooleanTensor with toStack joined along a new dimension
+     * <p>
+     * e.g. A, B, C = BooleanTensor.trues(4, 2)
+     * <p>
+     * BooleanTensor.stack(0, A, B, C) gives BooleanTensor.trues(3, 4, 2)
+     * <p>
+     * BooleanTensor.stack(1, A, B, C) gives BooleanTensor.trues(4, 3, 2)
+     * <p>
+     * BooleanTensor.stack(2, A, B, C) gives BooleanTensor.trues(4, 2, 3)
+     * <p>
+     * BooleanTensor.stack(-1, A, B, C) gives BooleanTensor.trues(4, 2, 3)
+     */
+    static BooleanTensor stack(int dimension, BooleanTensor... toStack) {
+        long[] shape = toStack[0].getShape();
+        int stackedRank = toStack[0].getRank() + 1;
```

Comment:
don't you need to check that all the tensors have the same shape?

Reply:
This is copy and pasted from `DoubleTensor`, which just forces the shape of the first element in `toStack` on the others

Reply:
I will add the suggested explicit checks to all our tensors, since numpy does the same.

Reply:
A new jira task is created.

Reply:
👍